timesketch/frontend/src/components/Common/TsIOCMenu.vue

```diff
@@ -0,0 +1,184 @@
+              <option v-for="option in IOCTypes" :value="option.type" :key="option.type">
+                {{ option.type }}
+              </option>
+            </b-select>
+            <b-button size="is-small" type="is-primary" @click="saveThreatIntel(params.data)">Add</b-button>
+          </b-field>
+          <div v-else>
+            <small>Already added to <router-link :to="{ name: 'Intelligence' }">Intelligence</router-link></small>
+          </div>
+        </section>
+      </template>
+    </TsContextMenu>
+  </div>
+</template>
+
+<script>
+import ApiClient from '../../utils/RestApiClient'
+import TsContextMenu from './TsContextMenu'
+import { SnackbarProgrammatic as Snackbar } from 'buefy'
+
+export default {
+  components: {
+    TsContextMenu,
+  },
+  props: ['attributeKey'],
+  name: 'TsIOCMenu',
+  data() {
+    return {
+      IOCTypes: [
+        { regex: /[0-9]{1,3}(\.[0-9]{1,3}\.)/g, type: 'ip' },
+        { regex: /[0-9a-f]{64}/gi, type: 'hash_sha256' },
+        { regex: /[0-9a-f]{40}/gi, type: 'hash_sha1' },
+        { regex: /[0-9a-f]{32}/gi, type: 'hash_md5' },
+      ],
+      iocColumns: [
+        { field: 'ioc', label: 'IOC' },
+        { field: 'type', label: 'Type' },
+      ],
+      selectedIOC: {},
+    }
+  },
+  methods: {
+    addFilter: function(field, value, operator) {
+      let chip = {
+        field: field,
+        value: value,
+        type: 'term',
+        operator: operator,
+        active: true,
+      }
+      this.$emit('addChip', chip)
+    },
+    getIOC: function(text) {
+      for (let iocType of this.IOCTypes) {
+        let matches = iocType.regex.exec(this.$attrs.text)
+        if (matches) {
+          return { ioc: this.$attrs.text, type: iocType.type }
+        }
+      }
+      return { ioc: this.$attrs.text, type: null }
+    },
+    isInIntelligence(ioc) {
+      const attributes = this.$store.state.meta.attributes
+      if (!attributes.intelligence_local) {
+        return false
+      }
+      if (attributes.intelligence_local.value.data.map(ioc => ioc.ioc).indexOf(ioc.ioc) >= 0) {
+        return true
+      }
+      return false
+    },
+    saveThreatIntel: function(ioc) {
+      ApiClient.getSketchAttributes(this.sketch.id).then(response => {
+        let attributes = response.data
+        if (!attributes.intelligence_local) {
+          attributes.intelligence_local = { ontology: 'intelligence', value: { data: [] } }
+        }
+
+        if (attributes.intelligence_local.value.data.map(ioc => ioc.ioc).indexOf(ioc.ioc) >= 0) {
+          return
+        }
+        attributes.intelligence_local.value.data.push(ioc)
+        ApiClient.addSketchAttribute(
+          this.sketch.id,
+          'intelligence_local',
+          attributes.intelligence_local.value,
+          'intelligence'
+        ).then(() => {
+          Snackbar.open({
+            message: 'Attribtue added succesfully',
+            type: 'is-white',
```

Comment:
Let's ceck dark mode here as well. But we can also leave it, seeing that I'm doing an overhaul of the dark mode in an upcoming PR